Es lint config

AndgelBrissaud opened the pull request.

Commit:
Ajout ESLint config
Ajout Logger.tools
Ajout ready.event.ts

utilisation de DayJs pour les dates

Commit:
ajout de commentaire dans le codes

AndgelBrissaud committed:
Merge branch 'main' into ESLint-config

PetitGens reviewed (commented):
Réponds à ça pour l'instant stp.

PetitGens commented on package.json:
Tu peux enlever stp ?

AndgelBrissaud replied:
installation due a ESLint

PetitGens replied:
C'est pas résolu, il faut enlever typescript de la prod. On en a pas besoin.

AndgelBrissaud commented on src/tools/logger.tool.ts:
```diff
@@ -0,0 +1,146 @@
+/**
+ * Define the available log levels in the application.
+ */
+export type LogLevel = 'log' | 'error' | 'warn' | 'debug' | 'verbose' | 'fatal';
```
j'avoues ils sont pas dans l'odre...
le logger ca uniformise les log, ca donne la date et l'heure du log systématiquement. 
c'est aussi pratique pour le dev que pour la prod.
mais pour le dev il y a debug. on peut ajouter une variable au .env et faire en sorte que debug de s'affiche que quand une variable est a true.

PetitGens replied:
Ouais, ça me va si c'est pas affiché par défaut.

Tu m'expliqueras plus tard les subtilités entre les niveaux au pire.

Commit:
correction typescript retrait de la dependence et logger remise dans l'ordre des type de log

Commit:
correction erreur dependence dev -> prod

PetitGens commented:
Je reviens plus tard après avoir testé le linter et tout.

PetitGens committed:
fix: typage des intéractions pour les handler de commandes

Refs: #3

PetitGens committed:
chore: npm i

Refs: #3

PetitGens committed:
chore: correction de l'indentation avec ESLint

Refs: #3

PetitGens committed:
chore: suppression commentaire TODO car réalisé

Refs: #3

PetitGens committed:
chore: remplacement de console.x par logger.x (utilisation du logger amélioré d'Andgel)

Refs: #3

PetitGens reviewed (changes requested):
C'est très bien, mais on pourrait peut-être : 

- [x] Faire en sorte que le logger accepte unknown en argument.
- [x] Déplacer les évènement de listeners.ts vers le dossier events
- [x] Fusionner les deux évènements ready
- [x] Faire en sorte d'enregistrer tous les écouteurs évènements dans le index.ts (s'inspirer de ce qui est fait pour les commandes)

PetitGens committed:
chore: formattage

Refs: #3

PetitGens committed:
chore: ESLint ignore les dossiers node_modules et dist

Refs: #3

PetitGens committed:
chore(linter): règle pour les nouvelles lignes en fin de fichier

Refs: #3

PetitGens committed:
chore: ligne vide en fin de fichier

Refs: #3

PetitGens committed:
chore: mise en place d'une commande "npm run watch"

Cette commande permet de compiler le bot automatiquement quand on change
un fichier dans le dossier src. De plus, à chaque compilation, le bot
est également relancé.

Cette commande est idéale pour compiler le projet pendant le
développement.

PetitGens committed:
chore: lancement d'ESLint en CICD

Refs: #6

PetitGens committed:
chore: utilisation de l'action 'eslint-annotate-action' pour remplacer un paquet avec des dépréciations

Refs: #6

PetitGens committed:
chore: corrections du workflow

- Noms un peu mieux pour les steps.
- Formattage de la commande (merci copilot)

Refs: #6

PetitGens committed:
chore: backslash pour copilot

PetitGens committed:
Merge pull request #8 from basse-cour/6-lancer-eslint-en-cicd

6 lancer eslint en cicd

Commit:
correction PR

Commit:
correction PR

AndgelBrissaud commented:
Il reste a tester le tout

PetitGens commented on src/scripts/deployEvent.script.ts:
À quoi sert ce script ?

PetitGens commented on src/scripts/deployCommands.script.ts:
```diff
@@ -0,0 +1,84 @@
+
+  let commandFiles: string[] = [];
+  try { 
+    const commandsDir = join(__dirname, "../commands");
+    commandFiles = readdirSync(commandsDir).filter((file) =>
+      file.endsWith(".command.ts")
+    );
```
On avait pas dit qu'on utilisait la constante COMMANDS ?

PetitGens commented:
Stp @AndgelBrissaud, tu pourrais un peu plus séparer tes commits pour les prochaines fois ? 😅 

C'est un peu dur à revoir quand tout est fait en même temps. Surtout quand tu renommes ou déplaces des fichiers ; ça il faut le faire dans un commit à part.
